zebra-chain/src/treestate/note_commitment_tree.rs

```diff
@@ -9,14 +9,53 @@
 //! append-only.
 //!
 //! A root of a note commitment tree is associated with each treestate.
+
 #![allow(clippy::unit_arg)]
+#![allow(dead_code)]
 
 use std::{fmt, io};
 
+use bitvec::prelude::*;
 #[cfg(test)]
 use proptest_derive::Arbitrary;
 
-use crate::serialization::{SerializationError, ZcashDeserialize, ZcashSerialize};
+use crate::{
+    commitments::sapling::pedersen_hash_to_point,
+    serialization::{SerializationError, ZcashDeserialize, ZcashSerialize},
+};
+
+/// Pedersen Hash Function
+///
+/// https://zips.z.cash/protocol/protocol.pdf#concretepedersenhash
+#[allow(non_snake_case)]
+fn pedersen_hash(domain: [u8; 8], M: &BitVec<Lsb0, u8>) -> jubjub::Fq {
+    jubjub::AffinePoint::from(pedersen_hash_to_point(domain, M)).get_u()
+}
```

Comment:
Should this maybe be grouped with the pedersen_hash_to_point function?

Reply:
The only thing that uses `pedersen_hash` is `merkle_crh_sapling`, which is for the note commitment tree, which is why it's over here. However, when things get reorg'd to `crate::sapling::*`, I'd like to move pretty much all these lower level hash functions into a common module, so, I think moving it to be with the ~rest in anticipation of that is fine.